xdsl/irdl/declarative_assembly_format.py

```diff
@@ -1011,7 +1011,9 @@ def parse(self, parser: Parser, state: ParsingState) -> bool:
         current_successor = parser.parse_optional_successor()
         while current_successor is not None:
             successors.append(current_successor)
-            current_successor = parser.parse_optional_successor()
+            if not parser.parse_optional_punctuation(","):
```

Comment:
I feel there's already a parser function for comma separated lists

Reply:
See `VariadicOperandVariable`. To be honest it doesn't seem like a great improvement over what is written here.